test/unit/Cardano/Wallet/CoinSelection/FeeSpec.hs

```diff
@@ -313,10 +528,69 @@ propReducedChanges drg (ShowFmt (FeeProp (CoinSelProp utxo txOuts) utxo' (fee, d
     adjust s = left show $ fst $ withDRG drg $ runExceptT $
         adjustForFees feeOpt utxo' s
 
+propFeeEstimation
+    :: FeeProp
+    -> Property
+propFeeEstimation (FeeProp (CoinSelProp utxo txOuts) _ _) = do
+    isRight selection ==> let Right s = selection in prop s
+  where
+    prop coinSel@(CoinSelection inps outs _) = do
+        let (Fee estFee) = estimateCardanoFee cardanoPolicy coinSel
+
+        -- | Make a Hash from a Base16 encoded string, without error handling.
+        let hash16 :: ByteString -> Hash a
+            hash16 = either bomb Hash . convertFromBase Base16
+                where
+                    bomb msg = error ("Could not decode test string: " <> msg)
+
+        -- | Make an Address from a Base58 encoded string, without error handling.
+        let addr58 :: ByteString -> Address
+            addr58 = maybe (error "addr58: Could not decode") Address
+                . decodeBase58 bitcoinAlphabet
+
+        let inputId0 = hash16 "60dbb2679ee920540c18195a3d92ee9be50aee6ed5f891d92d51db8a76b02cd2"
+        let address0 = addr58 "DdzFFzCqrhsug8jKBMV5Cr94hKY4DrbJtkUpqptoGEkovR2QSkcA\
+                              \cRgjnUyegE689qBX6b2kyxyNvCL6mfqiarzRB9TRq8zwJphR31pr"
+        let pkWitness = "\130X@\226E\220\252\DLE\170\216\210\164\155\182mm$ePG\252\186\195\225_\b=\v\241=\255 \208\147[\239\RS\170|\214\202\247\169\229\205\187O_)\221\175\155?e\198\248\170\157-K\155\169z\144\174\ENQhX@\193\151*,\NULz\205\234\&1tL@\211\&2\165\129S\STXP\164C\176 Xvf\160|;\CANs{\SYN\204<N\207\154\130\225\229\t\172mbC\139\US\159\246\168x\163Mq\248\145)\160|\139\207-\SI"
+        let txIns = zipWith TxIn (replicate (length inps) inputId0) [0..]
+        let txOuts' = zipWith TxOut (replicate (length outs) address0) (map coin outs)
```

Comment:
This is missing the change too :point_up:

Reply:
corrected